pilot/pkg/model/validation.go

```diff
@@ -2478,18 +2478,27 @@ func appendErrors(err error, errs ...error) error {
 // ValidateNetworkEndpointAddress checks the Address field of a NetworkEndpoint. If the family is TCP, it checks the
 // address is a valid IP address. If the family is Unix, it checks the address is a valid socket file path.
 func ValidateNetworkEndpointAddress(n *NetworkEndpoint) error {
-	switch n.Family {
+	return validateAddress(n.Family, n.Address)
+}
+
+// ValidateIstioEndpointAddress checks the Address field of a IstioEndpoint. If the family is TCP, it checks the
+// address is a valid IP address. If the family is Unix, it checks the address is a valid socket file path.
+func ValidateIstioEndpointAddress(ie *IstioEndpoint) error {
+	return validateAddress(ie.Family, ie.Address)
+}
+
+// validateAddress checks ip or unix address
+func validateAddress(family AddressFamily, address string) error {
+	switch family {
 	case AddressFamilyTCP:
-		ipAddr := net.ParseIP(n.Address) // Typically it is an IP address
+		ipAddr := net.ParseIP(address) // Typically it is an IP address
 		if ipAddr == nil {
-			if err := ValidateFQDN(n.Address); err != nil { // Otherwise could be an FQDN
```

Comment:
why remove fqdn?

Reply:
As described in [envoy](https://www.envoyproxy.io/docs/envoy/latest/api-v2/api/v2/endpoint/endpoint.proto#envoy-api-msg-endpoint-endpoint), for EDS, the address of endpoint is expected to be a direct IP address. Actually, if I create a serviceentry below, whose host is  already existed in cluster 
```
apiVersion: networking.istio.io/v1alpha3
kind: ServiceEntry
metadata:
  name: internal-svc-mesh
  namespace: bar
spec:
  hosts:
  - httpbin.bar.svc.cluser.local
  location: MESH_INTERNAL
  ports:
  - number: 80
    name: http
    protocol: HTTP
  resolution: DNS
```
envoy will report a error: 
```
error_detail:<code:13 message:"malformed IP address: httpbin.bar.svc.cluster.local. Consider setting resolver_name or setting cluster type to 'STRICT_DNS' or 'LOGICAL_DNS'" >
```

Reply:
Yeah but what if you don't have `httpbin.bar.svc.cluser.local` already in your cluster? Then this service entry is perfectly valid

Reply:
Under normal circumstances, it works well. But I think such case should be taken into consideration, or the whole mesh would crash.

Reply:
sure but this breaks a bunch as well. How can I define a service entry for an external service now ?

Reply:
I have run e2e locally, and there is no impact on external service